### PR TITLE
Extensions

### PR DESCRIPTION
**Pull request by divyaamunugama**
## Purpose
$subject

### CODE REVIEW

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 1]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
-|[protected void processAuthenticationResponse(HttpServletRequest request,HttpServletResponse response, AuthenticationContext context)](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticator.java#L169-L171)|Implements the logic of the custom federated authenticator.|
+## Write the federated authenticator
+
+To write the federated authenticator:
+
+1. Create a `maven` project for the custom federated authenticator. Refer the following files
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:118]:
```suggestion
1. Create a `maven` project for the custom federated authenticator. Refer the following files:
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 2]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
+    - [pom.xml](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml) file used for the sample custom federated authenticator.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:119]:
```suggestion
    - The [pom.xml](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/pom.xml) file used for the sample custom federated authenticator.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 3]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
+    - [service component class](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/internal/CustomFederatedAuthenticatorServiceComponent.java) file to deploy in WSO2 Identity Server and register it as a federated authenticator.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:120]:
```suggestion
    - The [service component class](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/internal/CustomFederatedAuthenticatorServiceComponent.java) file to deploy in WSO2 Identity Server and register it as a federated authenticator.
```

Comment (nilminiwso2) [reply en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:120]: Register `what` as a federated authenticator?

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 4]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
+
+2. Write the [custom federated authenticator](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticator.java) by extending the
+[AbstractApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java) class and implementing the
+[FederatedApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/FederatedApplicationAuthenticator.java) class.
+
+!!! info
+    You can find a custom federated authenticator [here](https://github.com/wso2/samples-is/tree/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator) for your reference.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:127]:
```suggestion
    See the sample [custom federated authenticator](https://github.com/wso2/samples-is/tree/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator) for your reference.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 5]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
+    | [protected void processAuthenticationResponse(HttpServletRequest request,HttpServletResponse response, AuthenticationContext context)](https://github.com/wso2/samples-is/blob/master/authenticators/components/org.wso2.carbon.identity.sample.federated.authenticator/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticator.java#L169-L171) | Implements the logic of the custom federated authenticator.    |
 
 ---
 
-## Deploy the custom federated authenticator in WSO2 IS
+## Deploy the authenticator
+
+To deploy the custom federated authenticator:
+
+1. Navigate to the root of your project and compile the service by running the following command.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:145]:
```suggestion
1. Open a terminal, navigate to the root of your project, and compile the service by running the following command.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 6]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
 ---
 
 ## Configure the partner identity server
 
 In this guide, the partner identity server acts as the external system. 
-Therefore, it is required to run a second identity server instance as the partner identity server. 
-The partner identity server will be running on the same machine in a different port 
-by adding the following config to the `<PARTNER_IS_HOME>/repository/conf/deployment.toml` file.
+Therefore, it is required to run a second identity server instance as the partner identity server.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:158]:
```suggestion
Therefore, it is required to run a second WSO2 Identity Server instance as the partner identity server.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 7]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
-```
+### Set up partner IS
+Add the following config to the `<PARTNER_IS_HOME>/repository/conf/deployment.toml` file to eun the partner identity server  on the same machine in a different port.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:161]:
```suggestion
Add the following config to the `<PARTNER_IS_HOME>/repository/conf/deployment.toml` file to be able to run the partner identity server  on the same machine in a different port.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 8]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
-1. Access the management console (`https://<PARTNER_IS_HOST>:<PARTNER_IS_PORT>/carbon`) of the partner identity server.
+To register and configure the service provider on the partner IS:
 
-2. Navigate to **Main** > **Identity** > **Service Providers** > **Add**.
+1. On the management console (`https://<PARTNER_IS_HOST>:<PARTNER_IS_PORT>/carbon`) of the partner identity server, go to **Main** > **Identity** > **Service Providers** > **Add**.
 
-    <img src="{{base_path}}/assets/img/fragments/add-service-provider.png" width="300" alt="Service provider option in Main menu" />
-    
-3. Enter a **Service Provider Name**. Optionally, enter a **Description**.
+2. Enter `pickup-dispatch` as the **Service Provider Name**, and click **Register**.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:177]:
```suggestion
2. Enter `pickup-dispatch` as the **Service Provider Name** and click **Register**.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 9]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
-    <img src="{{base_path}}/assets/img/fragments/register-service-provider.png" width="600" alt="Add new service provider basic information" />
-    
-4. Click **Register**.    
-   
-5. In the next view, expand **Inbound Authentication Configuration** > **OAuth/OpenID Connect Configuration** and click **Configure**.
+3. Expand the **Inbound Authentication Configuration > OAuth/OpenID Connect Configuration** section and, click **Configure**.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:179]:
```suggestion
3. Expand **Inbound Authentication Configuration > OAuth/OpenID Connect Configuration** and, click **Configure**.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 10]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
-6. Enter the **Callback Url** as `https://localhost:9443/commonauth`. 
+4. Enter the **Callback Url** as `https://localhost:9443/commonauth`.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:181]:
```suggestion
4. Enter `https://localhost:9443/commonauth` as the **Callback Url**.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 11]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
-7. Click **Add**. Note the **OAuth Client Key** and **OAuth Client Secret** that appear. 
+5. Click **Add**. Note that `client key` and `client secret` are generated.
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:183]:
```suggestion
5. Click **Add**. Note that the `client key` and `client secret` are generated.
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 12]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
 
-    !!! tip
-        To configure more advanced configurations, see [OAuth/OpenID Connect Configurations]({{base_path}}/guides/login/oauth-app-config-advanced). 
+6. Click **Update** to save the configurations.
 
-### Create a user
+!!! info "Download and deploy the application"
+    On the Partner Identity Server:
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:188]:
```suggestion
    On the partner identity server:
```

File: en/docs/references/extend/federation/write-a-custom-federated-authenticator.md [hunk 13]
```diff
@@ -111,148 +111,163 @@ authenticator.
...
+    - Download the [OIDC pickup-dispatch sample](https://github.com/wso2/samples-is/releases/download/v4.3.0/pickup-dispatch.war) application
```
Comment (nilminiwso2) [inline en/docs/references/extend/federation/write-a-custom-federated-authenticator.md:189]:
```suggestion
    - Download the [OIDC pickup-dispatch sample](https://github.com/wso2/samples-is/releases/download/v4.3.0/pickup-dispatch.war) application.
```

**Review by himeshsiriwardana — APPROVED** [review]

**Pull request merged**